### ISSUE TITLE
More like a question

### ISSUE THREAD

**Issue by subhrajitroy** [danielgerlag/workflow-core#117]
Is there an in built way that will let me wait for the Workflow to complete ?

As i understand the workflow will be added to a queue and status can be obtained by querying the persisted instance with workflow Id.
Beyond that is there a way a new API can be introduced that will let the invoker of the workflow wait till it is over ?

**Comment by germandb** [comment]
I would create a last step that have the only responsability of send a message to my system to let it know that the workflow steps are completed. 
In my project I resolve this using RabbitMQ  so my last step send a message to a queue and my application have a listener suscribed to that queue.

**Comment by subhrajitroy** [comment]
Thanks for the prompt reply. My use case is more like user or API consumer makes a call and that results in the workflow being triggered with the user waiting for the response. I understand that ideally the server could have just returned a 202 and continued with the workflow and as you suggest notify completion or error through an event. However the API consumers , essentially the mobile apps will not be changed for this , at least not in near future. Also I understand that this library may be the closest that we have been able to find that can co-ordinate a workflow , and in this particular case it is not going to be long running in that sense. I was wondering if I can use the fluent APIs to create the workflow and directly use the Workflow executor. Sorry to use this forum which I know is only for bugs.
Also I have a suggestion , from a project structure point of view , the workflow building and execution can be made different  projects so that they can be used as different  nuget packages. I found the fluent API really nicely done.
